common/lib/xmodule/xmodule/modulestore/courseware_index.py

```diff
@@ -64,13 +64,21 @@ def _fetch_item(item_location):
                 log.warning('Cannot find: %s', item_location)
                 return None
 
+            if groups_usage_info:
+                item.content_groups = []
+                for name, group in groups_usage_info:
+                    for module in group:
+                        if str(item_location) == str(module['usage_key_string']):
+                            item.content_groups.append(name)
             return item
 
-        def index_item_location(item_location, current_start_date):
+        def index_item_location(item_location, current_start_date, content_groups=None):
```

Comment:
You'll want to integrate in the same fashion into the latest changes for indexing.